### ISSUE TITLE
Support deployment behind a proxy

### ISSUE THREAD

**Issue by twm** [twm/yarrharr#308]
Right now Yarrharr will reject all requests when put behind a proxy because Django's [ALLOWED_HOSTS](https://docs.djangoproject.com/en/2.1/ref/settings/#std:setting-ALLOWED_HOSTS) will reject all requests.

Support deriving the original request origin from [proxy headers](https://httpd.apache.org/docs/2.4/mod/mod_proxy.html#x-headers), e.g.,:

```
X-Forwarded-For: <remote IP address>
X-Forwarded-Host: <server hostname>
X-Forwarded-Server: <server hostname>
```

This can be done right now at the Django level by setting [USE_X_FORWARDED_HOST](https://docs.djangoproject.com/en/2.1/ref/settings/#std:setting-USE_X_FORWARDED_HOST) to `True`, but that won't protect resources served by the Twisted parts of the app.

**Comment by twm** [comment]
See also https://twistedmatrix.com/trac/ticket/5807

**Comment by twm** [comment]
This was addressed adequately by #311. There are some todos in the code to add support for putting Yarrharr on a subpath, rather than the domain root, but I don't need that feature right now. I opened #936 specifically for the IPs in access logs.
